dnf5daemon-server/dbus/interfaces/org.rpm.dnf.v0.Base.xml

```diff
@@ -89,38 +89,18 @@ along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
         download_end:
         @session_object_path: object path of the dnf5daemon session
         @download_id: unique id of downloaded object (repo or package)
-        @status: libdnf5::repo::DownloadCallbacks::TransferStatus (0 - successful, 1 - already exists, 2 - error)
+        @transfer_status: libdnf5::repo::DownloadCallbacks::TransferStatus (0 - successful, 1 - already exists, 2 - error)
         @error: error message in case of failed download
 
         Downloading has ended.
     -->
     <signal name="download_end">
         <arg name="session_object_path" type="o" />
         <arg name="download_id" type="s" />
-        <arg name="status" type="i" />
+        <arg name="transfer_status" type="u" />
         <arg name="message" type="s" />
     </signal>
 
-    <!--
-        repo_key_import_request:
-        @session_object_path: object path of the dnf5daemon session
-        @key_id: PGP key id
-        @user_ids: User id
-        @key_fingerprint: Fingerprint of the PGP key
-        @key_url: URL of the PGP key
-        @timestamp: timestamp when the key was created
-
-        Request for repository key import confirmation.
-    -->
-    <signal name="repo_key_import_request">
```

Comment:
There is a bug in callbacks.cpp: KeyImportRepoCB::repokey_import() emits
a dnfdaemon::SIGNAL_REPO_KEY_IMPORT_REQUEST signal in wrong
dnfdaemon::INTERFACE_BASE interface. On the other hand, the signal is correctly registered in services/repo/repo.cpp as dnfdaemon::INTERFACE_REPO.

I guess we want to change callbacks.cpp. Technically, it's will break D-Bus API as we will move the singnal into a different interface. Do you want to resolve the callbacks.cpp bug in a separate issue targetting dnf-5.2.0.0 branch, or do you want to sweep it under the rug in this pull request?

Reply:
Hm, this is unfortunate. The signal was originally registered on Repo interface, but in base.cpp. So I just moved the registration to repo.cpp, which was a wrong decision. Thanks for noticing it!
I think it's safer to keep the signal on Base interface as the clients need to handle it in order to correctly import repo keys. I'd rather not disrupt them.